test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/DeepInsertE2ETests.cs

```diff
@@ -181,6 +181,53 @@ public void DeepInsertAnEntry_WithOneLevelOfNesting()
             Assert.Equal(1001, returnedCar.ID);
         }
 
+        [Fact]
+        public async Task DeepInsertAsyncAnEntry_WithOneLevelOfNesting()
+        {
+            var expectedResponse = "{\"@context\":\"http://localhost:5000/$metadata#Persons(Cars())/$entity\",\"ID\":100,\"Name\":\"Person 100\",\"Cars\":[{\"ID\":\"1001\",\"Name\":\"Car 1001\"}]}";

```

Comment:
It looks every test cases have the same logic to setup the data, for example from line 187 to 211?  maybe consider to create a helper method to setup?

Reply:
I have refactored the tests